packages/devtools_app/lib/src/app.dart

```diff
@@ -284,21 +284,23 @@ class DevToolsAppState extends State<DevToolsApp> with AutoDisposeMixin {
               .where((p) => embed && page != null ? p.screenId == page : true)
               .where((p) => !hide.contains(p.screenId))
               .toList();
+          final connectedToFlutterApp =
+              serviceConnection.serviceManager.connectedApp?.isFlutterAppNow ??
+                  false;
           return MultiProvider(
             providers: _providedControllers(),
             child: DevToolsScaffold(
               embed: embed,
               page: page,
               screens: screens,
               actions: [
-                if (connectedToVmService)
-                  // TODO(https://github.com/flutter/devtools/issues/1941)
-                  if (serviceConnection
-                          .serviceManager.connectedApp?.isFlutterAppNow ??
-                      false) ...[
-                    const HotReloadButton(),
-                    const HotRestartButton(),
-                  ],
+                if (connectedToVmService) ...[
+                  // TODO(kenz): do we need to hide this for Dart Web?
+                  HotReloadButton(
+                    callOnVmServiceDirectly: !connectedToFlutterApp,
+                  ),
```

Comment:
@elliette does Dart Web have hot reload functionality like dart server apps?

Reply:
It depends - for Flutter, hot reload simply triggers hot restart: https://github.com/flutter/flutter/issues/39848 For acx, hot reload is supported.

Reply:
Okay, so how can we make sure this Hot Reload button calls the right path for ACX? For ACX is it implemented through a service extension like for Flutter? Or for ACX do we need to call `service?.reloadSources` directly like we are now doing in this PR for Dart CLI apps?

Reply:
With the latest commit, I hid this button for dart web apps. This is not a regression since we were only showing this button for flutter apps before this PR. I've added a TODO to show this button for Dart web apps when supported, but we need to figure out